File: dotnet/cafdemalihapa/Applications/DirectoryManagement/Creation.cs
using cafdemalihapa.Directories.Hosting.Feature.BackEnd;
using cafdemalihapa.Directories.Hosting.Feature.BackEnd.BackEndPrimary;
using cafdemalihapa.Directories.Hosting.Feature.BackEnd.BackEndSecondary;
using cafdemalihapa.Directories.Hosting.Feature.Calls;
using cafdemalihapa.Directories.Hosting.Feature.Data;
using cafdemalihapa.Directories.Hosting.Feature.FrontEnd;
using cafdemalihapa.Directories.Hosting.Feature.FrontEnd.FrontEndSecondary;
using cafdemalihapa.Directories.Hosting.Feature.FrontEnd.FrontEndPrimary;
using cafdemalihapa.Directories.Hosting.Feature.NotesAndMessages;
using cafdemalihapa.Directories.Hosting.Feature.Tools;
using cafdemalihapa.Directories.Hosting.Feature.WebLinks;

namespace cafdemalihapa.Applications.DirectoryManagement
{
    public class Creation
    (
        DataDirectory dataDirectory,
        ToolsDirectory toolsDirectory,
        BackEndPrimaryDirectory backEndPrimaryDirectory,
        BackEndSecondaryDirectory backEndSecondaryDirectory,
        CallsDirectory callsDirectory,
        FrontEndDirectory frontEndDirectory,
        FrontEndPrimaryDirectory frontEndPrimaryDirectory,
        FrontEndSecondaryDirectory frontEndSecondaryDirectory,
        NotesAndMessagesDirectory notesAndMessagesDirectory,
        WebLinksDirectory webLinksDirectory
    )
    {
        public void Create()
        {
            if (!Commands.Get().IsCreate())
            {
                return;
            }
            CreateDirectories();
            CreateFiles();
        }

        public void CreateDirectories()
        {
            var directoriesToCreate = new List<Action>
            {
                toolsDirectory.Create,
                BackEndDirectory.Create,
                backEndPrimaryDirectory.Create,
                backEndSecondaryDirectory.Create,
                callsDirectory.Create,
                dataDirectory.Create,
                frontEndDirectory.Create,
                frontEndPrimaryDirectory.Create,
                frontEndSecondaryDirectory.Create,
                notesAndMessagesDirectory.Create,
                webLinksDirectory.Create,
            };

            foreach (var create in directoriesToCreate)
            {
                create();
            }
        }

        public void CreateFiles()
        {
            var filesToCreate = new List<Action>
            {
                frontEndPrimaryDirectory.CreateFiles,
                frontEndSecondaryDirectory.CreateFiles,
                notesAndMessagesDirectory.CreateFiles,
            };

            foreach (var createFiles in filesToCreate)
            {
                createFiles();
            }
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Applications/DirectoryManagement/Creation.fs
namespace cafdemwimu.console.Applications.DirectoryManagement

open cafdemwimu.console.Applications
open cafdemwimu.console.Directories.Hosting.Feature.BackEnd
open cafdemwimu.console.Directories.Hosting.Feature.BackEnd.BackEndPrimary
open cafdemwimu.console.Directories.Hosting.Feature.BackEnd.BackEndSecondary
open cafdemwimu.console.Directories.Hosting.Feature.BackEnd.BackEndTertiary
open cafdemwimu.console.Directories.Hosting.Feature.Calls
open cafdemwimu.console.Directories.Hosting.Feature.Data
open cafdemwimu.console.Directories.Hosting.Feature.FrontEnd
open cafdemwimu.console.Directories.Hosting.Feature.FrontEnd.FrontEndPrimary
open cafdemwimu.console.Directories.Hosting.Feature.FrontEnd.FrontEndSecondary
open cafdemwimu.console.Directories.Hosting.Feature.FrontEnd.FrontEndTertiary
open cafdemwimu.console.Directories.Hosting.Feature.NotesAndMessages
open cafdemwimu.console.Directories.Hosting.Feature.Tools
open cafdemwimu.console.Directories.Hosting.Feature.WebLinks

type Creation
    (
        dataDirectory: DataDirectory,
        toolsDirectory: ToolsDirectory,
        backEndPrimaryDirectory: BackEndPrimaryDirectory,
        backEndSecondaryDirectory: BackEndSecondaryDirectory,
        backEndTertiaryDirectory: BackEndTertiaryDirectory,
        callsDirectory: CallsDirectory,
        frontEndDirectory: FrontEndDirectory,
        frontEndPrimaryDirectory: FrontEndPrimaryDirectory,
        frontEndSecondaryDirectory: FrontEndSecondaryDirectory,
        frontEndTertiaryDirectory: FrontEndTertiaryDirectory,
        notesAndMessagesDirectory: NotesAndMessagesDirectory,
        webLinksDirectory: WebLinksDirectory
    ) =
    member this.Create() =
        if Commands.Get().IsCreate() then
            this.CreateDirectories()
            this.CreateFiles()

    member _.CreateDirectories() =
        let directoriesToCreate: (unit -> unit) list =
            [ (fun () -> toolsDirectory.Create())
              (fun () -> BackEndDirectory.Create())
              (fun () -> backEndPrimaryDirectory.Create())
              (fun () -> backEndSecondaryDirectory.Create())
              (fun () -> backEndTertiaryDirectory.Create())
              (fun () -> callsDirectory.Create())
              (fun () -> dataDirectory.Create())
              (fun () -> frontEndDirectory.Create())
              (fun () -> frontEndPrimaryDirectory.Create())
              (fun () -> frontEndSecondaryDirectory.Create())
              (fun () -> frontEndTertiaryDirectory.Create())
              (fun () -> notesAndMessagesDirectory.Create())
              (fun () -> webLinksDirectory.Create()) ]

        for create in directoriesToCreate do
            create ()

    member _.CreateFiles() =
        let filesToCreate: (unit -> unit) list =
            [ (fun () -> frontEndPrimaryDirectory.CreateFiles())
              (fun () -> frontEndSecondaryDirectory.CreateFiles())
              (fun () -> frontEndTertiaryDirectory.CreateFiles())
              (fun () -> notesAndMessagesDirectory.CreateFiles()) ]

        for createFiles in filesToCreate do
            createFiles ()
